### DISCUSSION TITLE
Nodes not working

### DISCUSSION THREAD

**Discussion by Naqua** [mikejac/node-red-contrib-google-smarthome#600] [Q&A]
Hello,

I followed the install instructions (checked everything more than twice), everything seems to work fine but when i put a node on a flow the node is visible in Google Home and online but i can't change the state. In node red the (lamp) node appears without a status (on/of). So it doesn't work but i am lost. And sometimes i get an error in the node red debug window but not always.

Any advice would be appreciated. Thanks in advance

![image](https://github.com/user-attachments/assets/9067bd4c-08a2-41bb-97cd-32545828f8ac)
![image](https://github.com/user-attachments/assets/50d66738-92b9-4c88-8139-fd6b816961d3)
![image](https://github.com/user-attachments/assets/c28a0579-45bc-415d-a9e9-8527bcb98376)

**Comment by Caprico85** [comment]
Hello @Naqua,

Ok, let's try to figure out the problem with the webserver first. This might solve the other the other problems as well.

Can you please set "Enable Node Debug" in the configuration. 

![image](https://github.com/user-attachments/assets/ea4228a7-8777-45dc-9116-5d7c01b66eda)

Then check Node-RED's debug logs. How you get to the debug log depends on how you installed Node-RED. If it is running as a Docker container, it will usually be `docker logs <container name>`. If running as a system service it might be `journalctl -u nodered`. Or it could be a file in `/var/log/`.

Then re-deploy your flows or restart Node-RED

If done correctly you should see lots of messages like 

```
GoogleSmartHome.constructor
Auth:loadAuthStorage(): data already persisted
Auth:loadAuthStorage(): me._authStorage = ...
SmartHome:Start(listen): using external SSL offload
SmartHome:Start(): registered routes:
SmartHome:Start(): url /oauth registered for get
...
```

Check if you see any warnings or errors in the log.

Can you also show us a screenshot of your configuration (where you set the ports and paths), please? Remember to anonymize the client ID and secret first.

**Reply by Naqua** [reply]
Hi, thanks for your kind response

Debug is/was on.

This is the log after deploy.
![image](https://github.com/user-attachments/assets/dc96ceed-8164-4506-95c4-f4002ebcf4c1)

The configuration of the management node (is that what you mean?)
![image](https://github.com/user-attachments/assets/a2b64c33-7917-48b1-89fe-7f7b2ff7359e)

Or this one:
![image](https://github.com/user-attachments/assets/7d985357-2a95-4a9d-a000-6ace5c533fa9)


I use caddy as reverse proxy. Followed the guide, seems to work accordingly.

Thanks again..

**Comment by Caprico85** [comment]
Hmm, so it is not the webserver. The configs and the log looks good. The webserver starts successfully, there is a successful sync request followed by the incoming SYNC call from Google and you can reach the check page. This would all not be possible if the webserver did not start.

The message `skipped request sync, the webserver is not running` can happen once during startup. I don't have it myself but I did see it on other people. I think we can ignore it for now.

So, we still have to get the states to update. Can you please leave the log enabled and then control your lamp in the app or via voice. There should be some log messages with one of them being
```HttpActions:httpActionsRegister(/smarthome): EXECUTE```
This is the incoming request from Google containing the actual command to control the light. If you don't see this, the request from Google is not reaching your host. Then [Google Cloud Logs](https://console.cloud.google.com/logs/query) would be your next stop.

A few lines later there should be the message
```google-smarthome:DeviceNode[Schreibtischlampe] .updateState set "on" to true```
This is our module decoding the payload to set the state.

**Reply by Naqua** [reply]
Hello and thanks again,

After pressing the on buton in google home app this is the logging:
![image](https://github.com/user-attachments/assets/de456d64-f250-4a21-b827-78231c86a482)

These are the corresponding Google Cloud Logs.
![image](https://github.com/user-attachments/assets/f5ee3335-da35-487f-907e-a35bb4447d10)

I can't make anything of this, your help is greatly appreciated :)

**Comment by Caprico85** [comment]
That's ... interesting. Your logs show that Google sends a QUERY request. This is, when you open the app and Google queries the current state of your devices. And judging by the logs from Google, there are EXECUTE request too when you press the lamp button. But the EXECUTE requests does not get through.

Can you please unfold the log entries in the Log Explorer, in case anything interesting is in there.

![image](https://github.com/user-attachments/assets/eb0e1f64-797b-4b57-91a6-0924b55ee721)

---

It could be related to local fulfillment. The EXECUTE command might reach your smart speaker but does not get forwarded to your host. Have you tried disabling local fulfillment, in case you enabled it? That is, on the Actions console, tab "Develop", remove both app.js scripts, delete all scan configurations and save.

![image](https://github.com/user-attachments/assets/89869371-4232-46e0-9f04-f8cd08282765)

In Node-RED, set Local Fulfillment to Disabled, save and deploy.

![image](https://github.com/user-attachments/assets/322d7578-8216-49c2-9074-ec863e92a2b0)

Then restart your smart speaker.

---

If this does not help, it might help to reset your smart home project.

On the Actions Console, on the tab "Test" click "Settings"

![image](https://github.com/user-attachments/assets/8972c081-e60e-4632-b800-3c9ba1d51814)

and disable "On device testing"

![image](https://github.com/user-attachments/assets/fe05998b-df8d-4d9a-a1cd-437f23a8af6e)

Click "Start testing". Then unlink and relink your account as explained [in the docs](https://github.com/mikejac/node-red-contrib-google-smarthome/blob/master/docs/setup_instructions.md#setup-account-linking)

**Reply by Naqua** [reply]
Hello and thanks again. Did everything you suggested.
Local fullfilment wasn't enabled anywhere. And never was.
Resetting the project and relinking didn't help either.
I tried a new fresh pi install wich gave exactly the same result.
Below are the folded out log entries.
![image](https://github.com/user-attachments/assets/07cde128-ecbe-4f97-b7af-c14dbaac62e6)

**Comment by Caprico85** [comment]
Sorry for the late response.

Can you please completely expand the log entries. There is a message deep in there if the action was successful. I hope in your case Google says if and why the action has failed.

![image](https://github.com/user-attachments/assets/71cde43b-8fb5-4a36-b51b-ffcb57465276)

**Reply by Naqua** [reply]
Thanks again for your reply, your assistance is greatly appreciated :)
Here are the completely unfolded log entries.

![image](https://github.com/user-attachments/assets/7e222e50-5d43-47dd-ad5b-636e60efaef8)
![image](https://github.com/user-attachments/assets/f7eedc37-811c-444c-9c70-b10571684709)
